### PR TITLE
Ensure components inherit content warning when they have a different scopecontent note

### PR DESCRIPTION
**Pull request by hackartisan**
- Update WC064 spec and add it as a content warning inheritance test case
- Ensure component inherits content warning

### CODE REVIEW

File: spec/features/traject/ead2_indexing_spec.rb [hunk 1]
```diff
@@ -464,20 +464,37 @@ def all_components(result)
...
+        context "when child had a non-warning scopecontent note" do
+          let(:fixture_path) do
+            Rails.root.join("spec", "fixtures", "aspace", "generated", "mss", "WC064.processed.EAD.xml")
+          end
+          it "still inherits the content warning, keeping its own scopecontent note" do
+            component = find_component(result, "WC064_c1")
+
+            # Ensure it's inheriting from its parent.
+            expect(component["scopecontent_combined_tsm"][0]).not_to eq "{}"
+            # expect(component["scopecontent_ssm"][0]).to include "harmful descriptions"
```
Comment (tpendragon) [inline spec/features/traject/ead2_indexing_spec.rb:494]: Delete?

File: lib/pulfalight/traject/ead2_component_config.rb [hunk 2]
```diff
@@ -547,40 +547,49 @@
...
-        end
+      parent_values = Array.wrap(parent.output_hash["#{selector}_combined_tsm"].clone)
+      # parent values is an array containing one stringified JSON hash
+      # with all the parent values
+      # extract just the content warning, keep it as a hash
+      parent_warning = parent_values.map do |parent_value|
+        ::JSON.parse(parent_value).slice("Content Warning")
+      end.reduce({}, :merge)
+
+      content.reverse_merge!(parent_warning)
```
Comment (tpendragon) [inline lib/pulfalight/traject/ead2_component_config.rb:576]: This is clever - my understanding is that "merge the two, but prefer the component's content warning if it exists" - yeah?

**Review by tpendragon — CHANGES_REQUESTED** [review]
Just one small delete. Looks great!

File: lib/pulfalight/traject/ead2_component_config.rb (hunk 2 above)
Comment (hackartisan) [reply lib/pulfalight/traject/ead2_component_config.rb:576]: Yep!

**Review by tpendragon — APPROVED** [review]

**Pull request merged**